src/ucp/core/ucp_ep.c

```diff
@@ -205,13 +241,14 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker,
 
     ep = ucp_worker_ep_find(worker, dest_uuid);
     if (ep != NULL) {
-        /* TODO: handle a case where:
-         * 1) the existing endpoint is incomplete
-         * 2) configuration (key) of existing endpoint does not match requested
-         *    parameters, for example, error handling level
-         */
-        *ep_p = ep;
-        status = UCS_OK;
+        status = ucp_ep_exist_adjust_params(ep, params);
+        if (status == UCS_OK) {
+            /* TODO: handle a case where the existing endpoint is incomplete */
```

Comment:
we can remove this TODO line or move the comment inside ucp_ep_exist_adjust_params, since that function will have to handle this case eventually..